CMakeCommon.cmake

```diff
@@ -190,6 +211,12 @@ function (generate_add_on_version_info outSemver)
 endfunction ()
 
 function (GenerateAddOnProject target acVersion devKitDir addOnSourcesFolder addOnResourcesFolder addOnLanguage)
+    # Capture optional arguments
+    set(options)
+    set(oneValueArgs additionalIncludesFolder additionalLibrary)
+    set(multiValueArgs)
+    cmake_parse_arguments(ARG "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
```

Comment:
This addition is very much unnecessary. The commands can be called on the target in `CMakeLists.txt`. Hardcoding `target_include_directories` to `SYSTEM PUBLIC` doesn't make sense. `target_link_libraries` is also using the legacy form where it's missing the requirement specifier, so I don't even know what the intention is with that line.